src/main/java/jenkins/plugins/slack/ActiveNotifier.java

```diff
@@ -154,7 +153,7 @@ public void completed(AbstractBuild r) {
     }
 
     private boolean skipOnMatrixChildren(AbstractBuild build) {
-        return build instanceof MatrixRun && !notifier.getMatrixTriggerMode().forChild;
+        return notifier.isMatrixProject(build) && !notifier.getMatrixTriggerMode().forChild;
```

Comment:
You're missing the other problem, getMatrixTriggerMode can return null :laughing:
`private MatrixTriggerMode matrixTriggerMode = MatrixTriggerMode.ONLY_CONFIGURATIONS;`
The should be no need for having it set on non-Matrix jobs either.

Reply:
Give me commit access than I have a fix for that :sweat_smile:

Reply:
done

Reply:
That should do it :sweat:

Reply:
I swear the code working before I merged it with yours :rofl:

Reply:
```
[2019-04-15T13:22:06.034Z] [ERROR] Possible null pointer dereference in jenkins.plugins.slack.ActiveNotifier.skipOnMatrixChildren(AbstractBuild) due to return value of called method [jenkins.plugins.slack.ActiveNotifier, jenkins.plugins.slack.ActiveNotifier] Dereferenced at ActiveNotifier.java:[line 157]Known null at ActiveNotifier.java:[line 157] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE

```

Reply:
I hate mocking! The isMatrixRun is being mocked out...

Reply:
SpotBugs on my end does not complain :confused:

Reply:
fails for me with 
```
mvn spotbugs:checl
```